core/src/main/java/com/linecorp/armeria/internal/AbstractHttp2ConnectionHandler.java

```diff
@@ -125,6 +125,27 @@ public void close(ChannelHandlerContext ctx, ChannelPromise promise) throws Exce
         super.close(ctx, promise);
     }
 
+    @Override
+    public void handlerAdded(ChannelHandlerContext ctx) throws Exception {
+        super.handlerAdded(ctx);
+        changeIdleStateHandlerToHttp2(ctx);
```

Comment:
This is for when a client upgrades the connection from http1.1 to http2. If there is a better way to deal with this situation, please do let me know.

Reply:
Probably better passing whether it's HTTP/2 in `HttpClientPipelinfConfigurator.finishSuccessfully()`?

Reply:
Sorry, did not get your comment. Let me elaborate the problem: 

For client, there is change to set `isHttp2` flag in `HttpClientPipelinfConfigurator.finishSuccessfully()`, please see [here](https://github.com/sivaalli/armeria/blob/5dc2d4a29b91cdd0cd8aabecc7b1d883bab7807c/core/src/main/java/com/linecorp/armeria/client/HttpClientPipelineConfigurator.java#L344) . 

On server side, the change is [here](https://github.com/sivaalli/armeria/blob/5c0ab40c6b7372ea9bea80302a8115d9f20e8cc5/core/src/main/java/com/linecorp/armeria/server/HttpServerPipelineConfigurator.java#L170). But when http client upgrades the connection using `upgrade` header, then `Http2ConnectionHandler` is added to pipeline and `HttpServerCodec` is removed. Since adding `Http2ConnectionHandler` to pipeline indicates a successful upgrade, I though to flip the state of `isHttp2` from `false` to `true` in `AbstractHttp2ConnectionHandler`

Now that I'm thinking about it, I think we should not have this logic in `AbstractHttp2ConnectionHandler` but rather in `Http2ServerConnectionHandler`. Please let me know what do you think 😄

Reply:
> Sorry, did not get your comment.

Ah, you were asking about the server-side. I thought the problem was in the client side. My bad :sweat_smile:  

> Now that I'm thinking about it, I think we should not have this logic in `AbstractHttp2ConnectionHandler` but rather in `Http2ServerConnectionHandler`. Please let me know what do you think 😄 

+1